### PR TITLE
Fixed parameter union type.

### LINKED ISSUE #6: Type of `delegate(Element, string, EventType, Handler)` doesn't seem to work
These work in TypeScript:

```js
delegate(".base", ".clickable", "click", handler);
delegate(select.all(".base"), ".clickable", "click", handler);
This doesn't
```

But the `Single base element specified` overload doesn't:

```js
delegate(select(".base"), ".clickable", "click", handler);
```

```
      TS2345: Argument of type 'HTMLElement | null' is not assignable to parameter of type 'string | ArrayLike<Element>'.
  Type 'null' is not assignable to type 'string | ArrayLike<Element>'.
```

cc @nickytonline

### PR DESCRIPTION
**Pull request by nickytonline**
Fixes #6

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -115,7 +115,7 @@ function delegate<TElement extends Element = Element, TEvent extends Event = Eve
 
 // Single base element specified
 function delegate<TElement extends Element = Element, TEvent extends Event = Event>(
-	elements: EventTarget | Document,
+	elements: EventTarget | Document | HTMLElement,
```
Comment (nickytonline) [inline index.ts]: `select(...)` from your example has the following type

```javascript
declare function select<T extends HTMLElement = HTMLElement>(selectors: string, baseElement?: BaseElement): T | null;
```

so I simply added `HTMLElement` to the union type for the `elements` parameter. I don't think `Element`, `DocumentFragment` or `Node` are required in the union type, but if you want those ones as well, let me know.

Comment (bfred-it) [reply index.ts]: This shouldn’t be required, HTMLElement is already a EventTarget

Comment (bfred-it) [reply index.ts]: Sometimes I get lost in the "direction" of types (e.g. assign Element to a node variable and viceversa) but this seems straightforward:

```
HTMLElement extends Element extends Node extends EventTarget
```

If `delegate` accepts `EventTarget` it should accept anything that extends it, right?

Comment (bfred-it) [reply index.ts]: https://www.typescriptlang.org/play/#src=let%20target%3A%20EventTarget%3B%0D%0A%0D%0Atarget%20%3D%20null%20as%20HTMLElement%3B

```ts
let target: EventTarget;
target = null as HTMLElement;
```

So `HTMLElement` should be assignable to type `EventTarget`

**Comment by bfred-it** [comment]
https://github.com/bfred-it/delegate-it/issues/6#issuecomment-490797740

**Pull request closed without merging**